rubrik_cdm/cluster.py

```diff
@@ -529,3 +529,124 @@ def refresh_vcenter(self, vcenter_ip, wait_for_completion=True, timeout=15):
             return self.job_status(api_request["links"][0]["href"])
 
         return self.post("v1", "/vmware/vcenter/{}/refresh".format(vcenter_id), timeout)
+
+    def update_proxy(self, host, protocol, port, username=None, password=None, timeout=15):
+        """Update the proxy configuration on the Rubrik cluster.
+
+        Arguments:
+            host {str} -- The IP address or FQDN of the proxy you wish to add.
+            protocol {str} -- The protocol of the proxy you wish to add.
+            port {int} -- The port of the proxy you wish to add.
+
+        Keyword Arguments:
+            username {str} -- The proxy username used for authentication. (default: {None})
+            password {str} -- The proxy password used for authentication. (default: {None})
+            timeout {int} -- The number of seconds to wait to establish a connection the Rubrik cluster before returning a timeout error. (default: {15})
+
+        Returns:
+            dict -- The full API response for `PATCH /node_management/proxy_config`
+        """
+
+        valid_protocols = [
+            'HTTP',
+            'HTTPS',
+            'SOCKS5']
+
+        if protocol not in valid_protocols:
+            sys.exit("Error: The protocol argument must be one of the following: {}.".format(
+                valid_protocols))
+
+        config = {}
+        config["host"] = host
+        config["protocol"] = protocol
+        config["port"] = port
+        config["username"] = username
+        config["password"] = password
+
+        self.log("proxy: Updated proxy configuration.")
```

Comment:
`self.log("proxy: Updated proxy configuration.")` should be `self.log("update_proxy: Updating proxy configuration.")` ... (use full function name and change Updated to Updating since they update hasn't actually happened at this point)